crates/red_knot_python_semantic/src/types.rs

```diff
@@ -255,11 +255,11 @@ pub enum Type<'db> {
     /// `Todo` would change the output type.
     Todo,
     /// A specific function object
-    Function(FunctionType<'db>),
+    FunctionLiteral(FunctionType<'db>),
     /// A specific module object
     Module(File),
```

Comment:
I wonder if it also makes sense to rename this to `ModuleLiteral`, to distinguish it from `Type::Instance(types.ModuleType)`

Reply:
Same thought here. Then I saw that `Module` was not part of `LiteralTypeKind`, so it's not treated as a literal in the `display` mod.

Reply:
I think how we display the type should probably be secondary to how we use and think about the type when it comes to naming. And there are already some confusing things about the `LiteralTypeKind` enum, e.g. that it (correctly) does not contain a variant for `LiteralString`.

Reply:
In #13257 I propose renaming `LiteralTypeKind` to `CondensedDisplayTypeKind` — it might be worth you doing that here or me pulling it out into a standalone PR. I think it's a clearer name

Reply:
> renaming `LiteralTypeKind` to `CondensedDisplayTypeKind`

I can do that.

And what do we do with `Type::Module`?

Reply:
> And what do we do with `Type::Module`?

I would vote to rename it to `ModuleLiteral` but otherwise leave it unchanged in this PR. I think if we rename `LiteralTypeKind` to `CondensedDisplayTypeKind`, it becomes more obvious that not all "literal" types really need to be included in that enum, just the ones that have condensed displays

Reply:
Done.